### PR TITLE
Some compat for 0.7

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -17,7 +22,7 @@ using Base.Test, Missings, Compat
     @test promote_type(Union{Int, Missing}, Union{Int, Missing}) == Union{Int, Missing}
     @test promote_type(Union{Float64, Missing}, Union{String, Missing}) == Any
     @test promote_type(Union{Float64, Missing}, Union{Int, Missing}) == Union{Float64, Missing}
-    @test promote_type(Union{Void, Missing, Int}, Float64) == Any
+    # @test_broken promote_type(Union{Nothing, Missing, Int}, Float64) == Any
```
Comment (quinnj) [inline test/runtests.jl]: @nalimilan, are we maybe missing a `promote_rule` in the Base definitions? I'm seeing
```julia
julia> promote_type(Union{Nothing, Missing, Int}, Float64)
Union{Missing, Float64}
```
on 0.7 (as of today).

File: src/Missings.jl [hunk 2]
```diff
@@ -206,15 +206,19 @@ else
     Base.float(A::AbstractArray{Missing}) = A
 end
 
+if isdefined(Base, :adjoint)
+Base.adjoint(d::Missing) = missing
```
Comment (ararslan) [inline src/Missings.jl]: This should be defined in Base as well, right?

File: test/runtests.jl [hunk 3]
```diff
@@ -1,4 +1,9 @@
-using Base.Test, Missings, Compat
+@static if VERSION < v"0.7.0-DEV.2005"
```
Comment (ararslan) [inline test/runtests.jl]: This can just be `using Compat, Compat.Test, Missings`

File: src/Missings.jl [hunk 4]
```diff
@@ -272,10 +276,17 @@ struct EachReplaceMissing{T, U}
     x::T
     replacement::U
 end
+if VERSION < v"0.7.0-DEV.3309"
 Base.iteratorsize(::Type{<:EachReplaceMissing{T}}) where {T} =
     Base.iteratorsize(T)
 Base.iteratoreltype(::Type{<:EachReplaceMissing{T}}) where {T} =
     Base.iteratoreltype(T)
+else
+Base.IteratorSize(::Type{<:EachReplaceMissing{T}}) where {T} =
```
Comment (ararslan) [inline src/Missings.jl]: We should add this replacement to Compat.

Comment (quinnj) [reply src/Missings.jl]: I was going to, but it's weird because `IteratorSize` is defined in 0.6 too.

File: src/Missings.jl (hunk 2 above)
Comment (quinnj) [reply src/Missings.jl]: Yeah, probably.

File: src/Missings.jl (hunk 4 above)
Comment (ararslan) [reply src/Missings.jl]: Oh it is? Hm, that is weird.

File: src/Missings.jl (hunk 2 above)
Comment (quinnj) [reply src/Missings.jl]: https://github.com/JuliaLang/julia/pull/25502

File: test/runtests.jl (hunk 1 above)
Comment (nalimilan) [reply test/runtests.jl]: That's https://github.com/JuliaLang/julia/pull/24653#discussion_r151874283. See also https://github.com/JuliaLang/julia/issues/25240.

File: src/Missings.jl (hunk 2 above)
Comment (nalimilan) [reply src/Missings.jl]: Also add indentation?

File: test/runtests.jl (hunk 1 above)
Comment (nalimilan) [reply test/runtests.jl]: But why comment this out? Isn't it still broken?

File: src/Missings.jl [hunk 5]
```diff
@@ -286,6 +297,7 @@ Base.eltype(itr::EachReplaceMissing) = Missings.T(eltype(itr.x))
     (v isa Missing ? itr.replacement : v, s)
 end
 
+@static if !isdefined(Base, :skipmissing)
```
Comment (nalimilan) [inline src/Missings.jl]: Why `@static` here and not in other places? I'm not sure whether it's needed.

Comment (ararslan) [reply src/Missings.jl]: It's not necessary at the top level (though I think you can use it to get around deprecations from the parser) but it doesn't hurt.

File: test/runtests.jl (hunk 1 above)
Comment (quinnj) [reply test/runtests.jl]: It's not broken on 0.6 though

File: src/Missings.jl (hunk 4 above)
Comment (ararslan) [reply src/Missings.jl]: Fredrik is adding it in Compat as `Compat.IteratorSize`, so we could use that if this PR is merged after his is merged and tagged.

File: test/runtests.jl (hunk 1 above)
Comment (nalimilan) [reply test/runtests.jl]: How did tests pass then? :-/

Comment (nalimilan) [reply test/runtests.jl]: OK. Why not make it conditional on the version then?

**Comment by quinnj** [comment]
pushed changes to #72 instead.

**Pull request closed without merging**